pyaerocom/colocation/colocation_setup.py

```diff
@@ -488,25 +489,26 @@ def basedir_logfiles(self):
         return str(p)
 
     @model_validator(mode="after")
-    @classmethod
-    def validate_obs_config(cls, v: PyaroConfig):
-        if v is not None and cls.obs.config.name != cls.obs_id:
+    def validate_obs_config(self):
+        if self.obs_config is None:
+            return self
+        if self.obs_config.name != self.obs_id:
             logger.info(
-                f"Data ID in Pyaro config {v.name} does not match obs_id {cls.obs_id}. Setting Pyaro config to None!"
+                f"Data ID in Pyaro config {self.obs_config.name} does not match obs_id {self.obs_id}. Setting Pyaro config to None!"
             )
-            v = None
-        if v is not None:
-            if isinstance(v, dict):
+            self.obs_config = None
+        if self.obs_config is not None:
+            if isinstance(self.obs_config, dict):
                 logger.info("Obs config was given as dict. Will try to convert to PyaroConfig")
-                v = PyaroConfig(**v)
-            if v.name != cls.obs_id:
+                self.obs_config = PyaroConfig(**self.obs_config)
+            if self.obs_id.name != self.obs_id:
```

Comment:
According to the type definition obs_id is str or None which doesn't have a name property?

Reply:
Yes, good catch. was a typo, should be `self.obs_config.name`